integration-tests/aws/pom.xml

```diff
@@ -30,6 +30,10 @@
     <description>The camel integration tests</description>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-main</artifactId>
+        </dependency>
```

Comment:
camel-quarkus-core is supposed to be used by people who want to create their own "main" whereas camel-quarkus-main does provide a ready to use "main" so no, component extensions should not depends on main otherwise it won't be possible for people to create their own main

Reply:
What are the potential use cases for creating one's own "mains"?

Reply:
You want to have a better control about the context initialization, set-up and life cycle as example.

Reply:
Do you need this in Camel K? Do you think 80+% of Camel Quarkus users will need this?

Reply:
I am trying to understand the significance of this. In case this is meant to be a speciality for <20% of users, then I think we should figure out how not confuse the rest 80% of the users.